tools/system_libs.py

```diff
@@ -1800,6 +1803,10 @@ class libmimalloc(MTLibrary):
     '-DMI_MALLOC_OVERRIDE',
     # TODO: add build modes that include debug checks 1,2,3
     '-DMI_DEBUG=0',
+    # disable `assert()` in emmalloc
```

Comment:
```suggestion
    # disable `assert()` in mimalloc
```
But do we not want this in debug builds?

Reply:
If we do, I think that we can/should make that a separate change.

Reply:
I think the comment is correct in this case, mimalloc uses `mi_assert`, which is already disabled.
https://github.com/microsoft/mimalloc/blob/v2.1.7/include/mimalloc/types.h#L563-L569

However, we also want to disable the assertions in the underlying emmalloc allocator, which we build as part of mimalloc:
https://github.com/emscripten-core/emscripten/blob/a090ba89182812faed943b864fff8b7046a8e6fa/tools/system_libs.py#L1818-L1819
https://github.com/emscripten-core/emscripten/blob/a090ba89182812faed943b864fff8b7046a8e6fa/tools/system_libs.py#L1828
https://github.com/emscripten-core/emscripten/blob/a090ba89182812faed943b864fff8b7046a8e6fa/system/lib/emmalloc.c#L33-L35
https://github.com/emscripten-core/emscripten/blob/a090ba89182812faed943b864fff8b7046a8e6fa/system/lib/libc/musl/include/assert.h#L5-L9

Reply:
Oh, it's for the underlying allocator. In that case it sounds ok. But can it be a separate change, or was it necessary for some reason to do in this PR?

Reply:
There's no specific reason to address it in this PR; it was discovered by accident while working on this. Split into PR #23016.